test/e2e/service_test.go

```diff
@@ -81,3 +82,60 @@ func (data *TestData) createClusterIPServiceAndBackendPods(t *testing.T, name st
 
 	return svc, cleanup
 }
+
+// TestNodePortWindows tests NodePort Service on Windows Node. It is a temporary test to replace upstream Kubernetes one:
+// https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/test/e2e/windows/service.go#L42
+func TestNodePortWindows(t *testing.T) {
+	skipIfNoWindowsNodes(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	svcName := "agnhost"
+	if len(clusterInfo.windowsNodes) == 0 {
+		t.Fatalf("The cluster should have at least one Windows Node")
+	}
+	svcNode := nodeName(clusterInfo.windowsNodes[0])
+	svc, cleanup := data.createAgnhostServiceAndBackendPods(t, svcName, svcNode, corev1.ServiceTypeNodePort)
+	defer cleanup()
+	t.Logf("%s Service is ready", svcName)
+
+	clientName := "agnhost-client"
+	require.NoError(t, data.createAgnhostPodOnNode(clientName, nodeName(0)))
```

Comment:
add a comment to emphasize that unlike the upstream e2e test, the client Pod is scheduled on a Linux Node (it seems that the code makes an assumption that `nodeName(0)` is a Linux Node)

Reply:
Added.

Reply:
thanks, it may be better to use `controlPlaneNodeName()` instead of `nodeName(0)`

Reply:
I think it's better not to emphasize it is a control-plane node here. I added more comments to explain why we choose `nodeName(0)`. So, I think it is better to still use `nodeName(0)` just like last time:
https://github.com/antrea-io/antrea/blob/48bed7412f72927dfc17d69b02f0534072486ef0/test/e2e/service_test.go#L38